src/wallet/wallet.cpp

```diff
@@ -4320,374 +4533,339 @@ bool CWallet::ConvertList(std::vector <CTxIn> vecTxIn, std::vector <CAmount> &ve
-                if (!DummySignTx(txNew, setCoins)) {
-                    strFailReason = _("Signing transaction failed");
-                    return false;
-                }
+        assert(nFound == 1);
+    }
 
-                unsigned int nBytes = ::GetSerializeSize(txNew, SER_NETWORK, PROTOCOL_VERSION);
+    CAmount totalInputValue = 0;
+    CAmount totalOutputValue = 0;
+    for (const CTransparentTxout& txin: vInputTxs) totalInputValue += txin.GetValue();
+    for (CTxOut& txout: tx.vout) totalOutputValue += txout.nValue;
+    assert(totalInputValue == totalOutputValue + nFee);
+    assert(totalInputValue > 0);
 
-                if (nExtraPayloadSize != 0) {
-                    // account for extra payload in fee calculation
-                    nBytes += GetSizeOfCompactSize(nExtraPayloadSize) + nExtraPayloadSize;
-                }
+    if (GetBoolArg("-walletrejectlongchains", DEFAULT_WALLET_REJECT_LONG_CHAINS)) {
+        // We should probably not make these transactions in the first place, but throwing this exception instead is the
+        // backwards compatible behaviour.
+        size_t nTotalMempoolAncestors = 0;
+
+        for (const CTransparentTxout& txin: vInputTxs)
+            nTotalMempoolAncestors += txin.GetDepthInMempool();
+
+        if (nTotalMempoolAncestors >= GetArg("-limitancestorcount", DEFAULT_ANCESTOR_LIMIT))
+            throw std::runtime_error("mempool chain");
+    }
+}
+
+// Create a transaction to vecSend, placing it in wtxNew. reservekey is the keypool. nFeeRet will be populated with the
+// calculated transaction fee. If nChangePosInOut is not -1, the change output will be placed at that position. If
+// nChangePosInOut > vecSend.size(), we will fail. If creating a transaction fails, we will return false and set
+// strFailReason to a human-readable description of the failure. coinControl may be set or unset; if it is unset an
+// arbitrary transaction selection algorithm will be used. If coinControl->nMaxSize is set, transaction creation will
+// fail if the size of the transaction with MAXIMUM length signatures would exceed the transaction size. Actual
+// generated transactions may produce signatures below this size, which may result in a transaction that would be
+// smaller than coinControl->nMaxSize being rejected for size reasons; this is because at the time the transaction is
+// made, we do not know how long the signatures will be. If sign is false, the transaction will not be signed; if it is
+// true, the wallet must be unlocked. nExtraPayloadSize should be set to the number of extra bytes in the transaction
+// outside inputs/outputs (ie. LLMQ-related things). If fUseInstantSend is true, we will consider both locked and
+// confirmed UTXOs to be eligible for input; if it is not, only confirmed UTXOs will be used as inputs.
+bool CWallet::CreateTransaction(const std::vector<CRecipient>& vecSend, CWalletTx& wtxNew, CReserveKey* reservekey,
```

Comment:
I don't think changing function signature (reserveKey parameter) is justified here, this function is inherited from bitcoin and subsequent merges with bitcoin PRs may create quite of lot of surprises